### PR TITLE
Fix StreambufWSMsg prepare (RIPD-1144).

### PR DESCRIPTION
**Pull request by miguelportilla**
Return value was set incorrect in some cases.
The buffer was not being prepared.
Add websocket log support.

### CODE REVIEW

File: src/ripple/server/impl/ServerHandlerImp.cpp [hunk 1]
```diff
@@ -271,12 +271,17 @@ ServerHandlerImp::onWSMessage(
                 sb.commit(boost::asio::buffer_copy(
                     sb.prepare(n), boost::asio::buffer(p, n)));
             });
+        JLOG(m_journal.trace())
+            << "Websocket sending '" << jvResult << "'";
```
Comment (sublimator) [inline src/ripple/server/impl/ServerHandlerImp.cpp]: noice, some logging

Comment (sublimator) [inline src/ripple/server/impl/ServerHandlerImp.cpp]: I believe Json::stream may be a co-conspirator here. I've not investigated deeply, but after logging the json tokens, and noticing progress, albeit extremely slowly, I suspect we must be doing a lot of needless reallocation/copying somehow. 

replacing Json::stream with
```c++
            auto const jrs = to_string(jr);
            auto n = jrs.length();
            void * p = (void *) jrs.c_str();
            sb.commit(boost::asio::buffer_copy(
                        sb.prepare(n), boost::asio::buffer(p, n)));
```

And I'm avoiding the stalls.

Next I tried preallocating the streambuf:
```c++
            auto const jrs = to_string(jr);
            beast::streambuf sb (jrs.length());
```

So ? We need some kind of heuristic to get a size hint ?  Or ?

In any case, later on we have code that looks like this:
```
            session->send(std::make_shared<
                StreambufWSMsg<decltype(sb)>>(std::move(sb)));
```

With the growth characteristics observed from afar (and the simple allocation hint tweak), this doesn't seem any better than just serialising to a string? How is this all supposed to work?

**Comment by vinniefalco** [comment]
The changes in this commit are okay, but we really could use the improvements offered by @sublimator

File: src/ripple/server/impl/ServerHandlerImp.cpp [hunk 2]
```diff
@@ -288,15 +288,11 @@ ServerHandlerImp::onWSMessage(
...
                 this->processSession(session, jc, jv);
-            beast::streambuf sb;
-            Json::stream(jr,
-                [&sb](auto const p, auto const n)
-                {
-                    sb.commit(boost::asio::buffer_copy(
-                        sb.prepare(n), boost::asio::buffer(p, n)));
-                });
-            JLOG(m_journal.trace())
-                << "Websocket sending '" << jr << "'";
```
Comment (sublimator) [inline src/ripple/server/impl/ServerHandlerImp.cpp]: We lost the logging ?

**Comment by vinniefalco** [comment]
Part of #1706

**Comment by vinniefalco** [comment]
:+1:

**Comment by nbougalis** [comment]
Merged as 154e90b1cad9e9d62431af15b6f2bdd792bc1aca

**Pull request closed without merging**